### PR TITLE
Bug 1904538: hack: filter SCCs for CMO role

### PR DESCRIPTION
**Pull request by s-urbaniak**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

/cc @openshift/openshift-team-monitoring

### CODE REVIEW

File: manifests/0000_50_cluster-monitoring-operator_02-role.yaml [hunk 1]
```diff
@@ -302,14 +294,6 @@ rules:
...
   - list
   - watch
-- apiGroups:
-  - security.openshift.io
-  resourceNames:
-  - node-exporter
-  resources:
-  - securitycontextconstraints
-  verbs:
-  - use
```
Comment (s-urbaniak) [inline manifests/0000_50_cluster-monitoring-operator_02-role.yaml:312]: i believe this is correct, there is no need to have the `node-exporter` SCC on the CMO pod.

File: manifests/0000_50_cluster-monitoring-operator_02-role.yaml [hunk 2]
```diff
@@ -113,11 +113,11 @@ rules:
   - securitycontextconstraints
   verbs:
   - create
-  - delete
   - get
   - list
-  - update
   - watch
+  - update
+  - delete
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_02-role.yaml:120]: /lgtm
not sure why it changed the ordering?

Comment (s-urbaniak) [reply manifests/0000_50_cluster-monitoring-operator_02-role.yaml:120]: :man_shrugging:

**Comment by s-urbaniak** [comment]
ok, the scc for node_exporter was indeed needed:
```
level=error msg=Cluster operator monitoring Degraded is True with UpdatingnodeExporterFailed: Failed to rollout the stack. Error: running task Updating node-exporter failed: reconciling node-exporter ClusterRole failed: creating ClusterRole object failed: clusterroles.rbac.authorization.k8s.io "node-exporter" is forbidden: user "system:serviceaccount:openshift-monitoring:cluster-monitoring-operator" (groups=["system:serviceaccounts" "system:serviceaccounts:openshift-monitoring" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:
```

**Comment by s-urbaniak** [comment]
I am wondering though, the message
```
is attempting to grant RBAC permissions not currently held:
```
is indicating that we are going to need that permission also with `nonroot`, let's see.

**Comment by simonpasquier** [comment]
> we are going to need that permission also with nonroot, let's see.

It would make sense. Another tactic would be to patch the kube-rbac-proxy's [Dockerfile](https://github.com/openshift/kube-rbac-proxy/blob/20ee6a36446e8434843a47c621283454d6340d56/Dockerfile.ocp#L20) to specify a numerical ID instead of `nobody`.

**Comment by s-urbaniak** [comment]
@simonpasquier agreed, i just want to quickly verify that this issue retains with just disabling the `nonroot` scc in e2e tests (but i am 99% sure anyways) as elevating CMO's permissions is not technically necessary, but it seems there is no way around it.

**Comment by s-urbaniak** [comment]
yeah, i think we have to modify CMO to be able to run as `nonroot`:
```
level=error msg=Cluster operator monitoring Degraded is True with UpdatingAlertmanagerFailed: Failed to rollout the stack. Error: running task Updating Alertmanager failed: reconciling Alertmanager ClusterRole failed: creating ClusterRole object failed: clusterroles.rbac.authorization.k8s.io "alertmanager-main" is forbidden: user "system:serviceaccount:openshift-monitoring:cluster-monitoring-operator" (groups=["system:serviceaccounts" "system:serviceaccounts:openshift-monitoring" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:
level=error msg={APIGroups:["security.openshift.io"], Resources:["securitycontextconstraints"], ResourceNames:["nonroot"], Verbs:["use"]}
```

**Comment by s-urbaniak** [comment]
closing out, as this is not the path we can follow unfortantely.

**Pull request closed without merging**